src/sources/external_input.ml

```diff
@@ -202,24 +207,31 @@ object (self)
           else
             begin
               if List.mem (`Read in_d) l then
-                begin
-                  try
-                    get_data in_d abg;
-                    (* Check that audio and video roughly get filled as the same speed. *)
-                    let lv = Frame.seconds_of_master (Generator.video_length abg) in
-                    let la = Frame.seconds_of_master (Generator.audio_length abg) in
-                    let d = abs_float (lv -. la) in
-                    if d -. vadiff_offset >= 0. then
-                      (
-                        vadiff_offset <- vadiff_offset +. vadiff_offset;
-                        let v, a = if lv >= la then "video", "audio" else "audio", "video" in
-                        self#log#severe
-                          "Got %f seconds more of %s than of %s. Are you sure \
-                           that you are producing the correct kind of data?" d v a
-                      )
-                  with
-                  | End_of_file -> raise (Finished ("Process exited.", restart))
-                end;
+                (* Do we need to read the headers? *)
+                if has_headers then
+                  (
+                    has_headers <- false;
+                    read_headers in_d
```

Comment:
How big are `AVI` headers on the average? The executing task is set as `Non_blocking` and waiting on a `read` signal. However, the headers read could potentially be blocking. I might chime in an adapt `Avi.Read.headers_simple` to work as a non-blocking duppy task as well.

Reply:
Headers are very "small" I would say (= a couple of metadata).

The thing is that
1. I precisely wanted to avoid duppyfying all the `Avi` module
2. pulling frames is also potentially blocking so it's not worse than before.

Reply:
Gotcha.